scripts/convert_and_dispatch_genotypes.py

```diff
@@ -257,32 +216,60 @@ def _run(self):
-            elif lims_sample.udf.get('User Sample Name') in self.geno_conv.sample_names:
-                self.info('Matching %s against user sample name %s', lims_sample.name, lims_sample.udf.get('User Sample Name'))
-                vcf_file = self.geno_conv.generate_vcf(lims_sample.udf.get('User Sample Name'), new_name=artifact.name)
-                genotyping_sample_used.append(lims_sample.udf.get('User Sample Name'))
             else:
                 self.info('No match found for %s', lims_sample.name)
                 invalid_lims_samples.append(lims_sample)
-            if vcf_file:
+            if found:
                 valid_samples.append(lims_sample)
-                if not self.no_upload:
-                    file = self.lims.upload_new_file(lims_sample, vcf_file)
-                    if file:
-                        lims_sample.udf['Genotyping results file id'] = file.id
-                        lims_sample.put()
+                vcf_file = self.geno_conv.generate_vcf(lims_sample.name)
+                nb_call = self.geno_conv.nb_calls(lims_sample.name)
+                output_arts = self.process.outputs_per_input(artifact, ResultFile=True)
+                # there should only be one
+                assert len(output_arts) == 1
+                # upload the number of calls to output
+                output_arts[0].udf[output_genotype_udf_number_call] = nb_call
+                output_arts[0].put()
+
+                # and the vcf file
+                file = self.lims.upload_new_file(lims_sample, vcf_file)
+                # increment the nb of tries
+                if not submitted_nb_genotype_tries in lims_sample.udf:
+                    lims_sample.udf[submitted_nb_genotype_tries] = 1
+                else:
+                    lims_sample.udf[submitted_nb_genotype_tries] += 1
+
+                if submitted_genotype_udf_number_call not in lims_sample.udf:
+                    # This is the first genotyping results
+                    lims_sample.udf[submitted_genotype_udf_number_call] = nb_call
+                    lims_sample.udf[genotype_udf_file_id] = file.id
+                elif lims_sample.udf.get(submitted_genotype_udf_number_call) and \
+                        nb_call > lims_sample.udf.get(submitted_genotype_udf_number_call):
+                    # This genotyping is better than before
+                    lims_sample.udf[submitted_genotype_udf_number_call] = nb_call
+                    lims_sample.udf[genotype_udf_file_id] = file.id
+                else:
+                    self.info(
+                        'Sample %s new genotype has %s call(s), previous genotype has %s call(s)',
+                        lims_sample.name,
+                        nb_call,
+                        lims_sample.udf[submitted_genotype_udf_number_call]
+                    )
+                # finally upload the submitted samples
+                lims_sample.put()
+
```

Comment:
Since we don't have the `elif` for user sample IDs at line 267 anymore, can't we just move this block inside the `if lims_sample.name in self.geno_conv.sample_names` instead of using the `found` variable?